### ISSUE TITLE
Additional Name Changes [CT/Model/API]

### ISSUE THREAD

**Issue by dih-cdisc** [cdisc-org/DDF-RA#117]
Previous ticket discussed additional name changes like code, potentially type and others? This ticket is to pick that work up for a subsequent sprint.

See [DDF RA #116](https://github.com/cdisc-org/DDF-RA/issues/116#issuecomment-1628213130)

**Comment by drewcdisc** [comment]
Additional items to consider:

- bcName
- bcPropertyName
- OrganizationName
- TimingDescription

**Comment by BSnoeijerCD** [comment]
Full team: discuss and propose standards for Name, Label, Description combinations + rationale. This can then be discussed and proposed at SME meeting. 
At end of this sprint or already before SME meeting of coming Tuesday?

For API: add (to) standard type to include in the API classes including the full combination. This will reduce the risk of inconsistencies and errors. (See Daves proposal in model split JIRA ticket.

**Comment by dih-cdisc** [comment]
Set of common attributes:

Already present:
- id
- name
- description

Additional, standard new ones:
- label
- type
- code

Current classes where an attribute might be added
- Indication class & 'name'
- StudyDesignPopulation class & 'name'
- Objective class & 'name'

Other potential renames
- Endpoint class & 'endpointPurpuseDescription' to 'purpose'
- [Set defined above ](https://github.com/cdisc-org/DDF-RA/issues/117#issuecomment-1638015016)

**Comment by drewcdisc** [comment]
@BSnoeijerCD Needs to be presented to the SME meeting on 2023-08-08 for review.

**Comment by drewcdisc** [comment]
To be clear: This means 5 renames and 3 new attributes. Anything beyond that goes in a separate PBI

**Comment by BSnoeijerCD** [comment]
> @BSnoeijerCD Needs to be presented to the SME meeting on 2023-08-08 for review.

@EMuhlbradt / @czwickl - If you have time you can present the proposals coming Tuesday and discuss with the groups. If you are not available I can do it for you.

**Comment by EMuhlbradt** [comment]
As of 2023-08-08 meeting, the team has agreed to change the following USDM element names: 
1. For BiomedicalConcept entity, change attribute name from 'bcName' to 'name'
2. For BiomedicalConceptProperty entity, change the attribute name from 'bcPropertyName' to 'name'
3. For Organization entity, change the attribute name from 'organizationName' to 'name'
4. For Timing entity, change the attribute name from 'timingDescription' to 'description'
5. For Endpoint entity, change the attribute name from 'endpointPurposeDescription' to 'purpose'

**Comment by BSnoeijerCD** [comment]
Thank you @EMuhlbradt and @MarkPAndrews 
@jared-schreibman-cdisc : seems you are good to go. Anything you are still missing for updating the API?

**Comment by MarkPAndrews** [comment]
@EMuhlbradt I got the updated XMI file but I found a few problems with the CT spreadsheet.
There is a `line feed` at the end of values in cells F208 and D216 that is causing the descriptions for those attributes to drop.  I removed them locally, but you should remove them in your copy.
I also noticed that there are 3 classes that you have a `description` attribute on, but the model doesn't. Can you confirm that they should be added?

- BiomedicalConceptCategory
- BiomedicalConceptSurrogate
- Encounter

If you can let me know today, I will try to get the changes made and into Git today.
CC: @BSnoeijerCD @jared-schreibman-cdisc @cupkes

**Comment by MarkPAndrews** [comment]
@EMuhlbradt @BSnoeijerCD  @czwickl  @dih-cdisc @jowencdisc I have pushed the Delta and Data Dictionary to the sprint-3-2 branch.  Please let me know if you have any concerns.

**Comment by EMuhlbradt** [comment]
I fixed the line feed issues in the CT deliverable. Also I confirmed that YES  there should be 'description' attributes attached to BiomedicalConceptCategory, BiomedicalConceptSurrogate, and Encounter entities. I'm not sure why those aren't in the model; they should have been there at the end of Phase 2. That being said, we updated the names in sprint 1 to 'description' so perhaps that is what is confusing the issue. As per sprint 1 of phase 3,
bcCategoryDescription changed to description
bcSurrogateDescription changed to description
encounterDescription changed to description

I found when I was checking those 3 items that I had accidentally put 'Description' into the sprint 1 product instead of 'description', so I updated the names to remove the upper-case D from the attribute names.

All of these changes should now be in the 3-2 branch.
